app/frontend/translations.js

```diff
@@ -1,43 +1,51 @@
     'Sending funds is not possible since there is not enough balance to pay the transaction fee',
   SendAmountPrecisionLimit: () => 'Invalid format: Maximum allowed precision is 0.000001',
   SendAmountIsTooBig: () =>
     `Invalid format: Amount cannot exceed ${printAda(Number.MAX_SAFE_INTEGER)}`,
+
   InvalidMnemonic: () => 'Invalid mnemonic, check your mnemonic for typos and try again.',
-  WalletInitializationError: ({message}) =>
-    `Error during wallet initialization${message ? `:  ${message}` : ''}`,
-  TransactionRejectedByNetwork: () => 'Submitting the transaction into Cardano network failed.',
-  TransactionRejected: ({sendResponse}) =>
-    sendResponse && sendResponse.message
-      ? sendResponse.message
-      : 'Transaction rejected while signing.',
-  TrezorRejected: () => 'Operation rejected by the Trezor hardware wallet.',
-  TransactionCorrupted: () => 'Transaction assembling failure.',
-  TransactionNotFoundInBlockchainAfterSubmission: ({sendResponse}) =>
-    `Transaction ${
-      sendResponse.txHash
-    } not found in blockchain after being submitted, check it later please.`,
-  UnknownCryptoProvider: ({cryptoProvider}) => `Uknown crypto provider: ${cryptoProvider}`,
-  NetworkError: () => 'Network connection failed. Please check your network connection.',
-  Error: () => `Unknown error, please contact support at ${ADALITE_SUPPORT_EMAIL}.`,
+
+  TransportOpenUserCancelled: ({message}) => `TransportCanceledByUser:${message}`,
+  TransportError: ({message}) => `TransportError:${message}`,
+  TransportStatusError: ({message}) => `TransportStatusError:${message}`,
+
+  TransactionRejectedByNetwork: () =>
+    'TransactionRejectedByNetwork:Submitting the transaction into Cardano network failed.',
+  TransactionRejectedWhileSigning: ({message}) =>
+    `Transaction rejected while signing${message ? `:  ${message}` : '.'}`,
+  TrezorRejected: () => 'TrezorRejected:Operation rejected by the Trezor hardware wallet.',
+  TransactionCorrupted: () => 'TransactionCorrupted:Transaction assembling failure.',
+  TransactionNotFoundInBlockchainAfterSubmission: ({txHash}) =>
+    `TransactionNotFoundInBlockchainAfterSubmission:
+    Transaction ${txHash ||
+      ''} not found in blockchain after being submitted, check it later please.`,
+  TrezorSignTxError: ({message}) => `TrezorSignTxError:${message}`,
+  LedgerOperationError: ({message}) => `LedgerOperationError:${message}`,
+  TxSerializationError: ({message}) => `TxSerializationError:${message}`,
+  CoinAmountError: () => 'CoinAmounError:Unsupported amount of coins.',
+  TrezorError: ({message}) => `TrezorError:${message}`,
+
+  CryptoProviderError: ({message}) => `CryptoProviderError:${message}`,
+  NetworkError: () =>
+    'NetworkError:Network connection failed. Please check your network connection.',
 }
 
 function getTranslation(code, params) {
   if (!translations[code]) {
     debugLog(`Translation for ${code} not found!`)
+    return null
```

Comment:
the function previously returned the 'code' it received as a parameter, now 'null' is returned in such a scenario. Now the calling functions assign null to errors instead of at least code. Is this correct behavior?

Reply:
I think this is a correct behavior. Previously, `code` was returned in case it didnt have a translation, but currently, this can never happen since every expected error has a translation, and if not, unexpectedErrorModal should be displayed which relies on this function returning `null` in that case.